semgrep/semgrep/cli.py

```diff
@@ -78,6 +81,13 @@ def cli() -> None:
         action="store_true",
         help="Validate configuration file(s). No search is performed.",
     )
+
+    config.add_argument(
+        "--lint",
```

Comment:
Should we have all this functionality under --validate?

Reply:
I wanted to keep this separate as (for the time being) something more experimental because:
- It's kind of slow
- Some of my lints might be wrong
- It will fire on lots of existing rules in semgrep-rules currently

Reply:
I could definitely imagine making it part of validate (or promoting individual lints to validations)

Reply:
what's an example of when it fires on semgrep-rules?

Reply:
```
lint: redundant patterns
  --> ../../semgrep-rules/python/flask/security/unsanitized_input.yaml:29
29 |           - pattern: |
30 |               $X = request.args.get(...)
31 |               ...
32 |               return flask.make_response(f"{$X}...")
  --> ../../semgrep-rules/python/flask/security/unsanitized_input.yaml:33
33 |           - pattern: |
34 |               $X = request.args.get(...)
35 |               ...
36 |               $R = flask.make_response(f"{$X}...")
= help: remove the first and delete `return` from the second
These patterns are redundant (return paired with assignment)
```

Reply:
Fixed lint errors in semgrep-rules: https://github.com/returntocorp/semgrep-rules/pull/522

Reply:
Idk if I would say these are redundant?

Reply:
Having those two rules in combination matches what you would match with:
```
34 |               $X = request.args.get(...)
35 |               ...
36 |               flask.make_response(f"{$X}...")
```
(you no longer need to specify `return` or `$A =` explicitly.)